projects/plugins/migration/src/js/components/hooks/use-migration-status.ts

```diff
@@ -0,0 +1,29 @@
+import { useState, useEffect, useCallback } from 'react';
+import { MigrationStatus } from '../migration/types';
+
+/**
+ * Get migration status
+ *
+ * @param {object} restApi - Configured restApi
+ * @returns {MigrationStatus} - MigrationStatus object
+ */
+export function useMigrationstatus( restApi ): MigrationStatus {
```

Comment:
Could we also check the users' connection first and decide to fetch it or not? If user hasn't connect to WordPress.com site yet, it's probably not efficiency to have it check every three second. What do you think?

Reply:
For sure, that’s a good point. I’ll add that part. Thanks!